src/geff/testing/data.py

```diff
@@ -0,0 +1,524 @@
+    ... )
+
+    # Using with GeffReader
+    >>> from geff import GeffReader
+    >>> store, props = create_simple_2d_geff()
+    >>> reader = GeffReader(store)
+    >>> graph = reader.read_nx()
+    >>> # graph is a networkx Graph ready for analysis
+"""
+
+from typing import Any, Literal, TypedDict, cast
+
+import networkx as nx
+import numpy as np
+import zarr
+import zarr.storage
+from numpy.typing import NDArray
+
+import geff
+
+DTypeStr = Literal["double", "int", "int8", "uint8", "int16", "uint16", "float32", "float64", "str"]
+Axes = Literal["t", "z", "y", "x"]
+
+
+class GraphAttrs(TypedDict):
+    nodes: NDArray[Any]
+    edges: NDArray[Any]
+    t: NDArray[Any]
+    z: NDArray[Any]
+    y: NDArray[Any]
+    x: NDArray[Any]
+    extra_node_props: dict[str, NDArray[Any]]
+    extra_edge_props: dict[str, NDArray[Any]]
+    directed: bool
+    axis_names: tuple[Axes, ...]
+    axis_units: tuple[str, ...]
+
+
+class ExampleNodeAxisPropsDtypes(TypedDict):
+    position: DTypeStr
+    time: DTypeStr
+
+
+def create_dummy_graph_props(
+    node_id_dtype: DTypeStr,
+    node_axis_dtypes: ExampleNodeAxisPropsDtypes,
+    directed: bool,
+    num_nodes: int = 5,
+    num_edges: int = 4,
+    extra_node_props: dict[str, DTypeStr] | None = None,
+    extra_edge_props: dict[str, DTypeStr] | None = None,
+    include_t: bool = True,
+    include_z: bool = True,
+    include_y: bool = True,
+    include_x: bool = True,
+) -> GraphAttrs:
+    """Create dummy graph properties for testing.
+
+    Args:
+        node_id_dtype: Data type for node IDs
+        node_axis_dtypes: Dictionary specifying dtypes for node axis properties (space and time)
+        directed: Whether the graph is directed
+        num_nodes: Number of nodes to generate
+        num_edges: Number of edges to generate
+        extra_node_props: Dict mapping property names to dtypes for extra node properties
+        extra_edge_props: Dict mapping property names to dtypes for extra edge properties
+        include_t: Whether to include time dimension
+        include_z: Whether to include z dimension
+        include_y: Whether to include y dimension
+        include_x: Whether to include x dimension
+
+    Returns:
+        Dictionary containing all graph properties
+    """
+    # Build axis_names and axis_units based on which dimensions to include
+    axis_names_list = []
+    axis_units_list = []
+
+    if include_t:
+        axis_names_list.append("t")
+        axis_units_list.append("s")
+    if include_z:
+        axis_names_list.append("z")
+        axis_units_list.append("nm")
+    if include_y:
+        axis_names_list.append("y")
+        axis_units_list.append("nm")
+    if include_x:
+        axis_names_list.append("x")
+        axis_units_list.append("nm")
+
+    axis_names: tuple[Axes, ...] = cast("tuple[Axes, ...]", tuple(axis_names_list))
+    axis_units = tuple(axis_units_list)
+
+    # Generate nodes with flexible count
+    if node_id_dtype == "str":
```

Comment:
We are going to remove the option for string node ids from the spec so you can just get rid of this